### PR TITLE
feat: now search callback response will emit SearchRequest after every search request is sent

### PR DESCRIPTION
**Pull request by zyf0330**

### CODE REVIEW

**Comment by zyf0330** [comment]
#728

File: docs/client.md [hunk 1]
```diff
@@ -276,15 +276,15 @@ containing the following fields:
...
-codes like the other APIs.  You'll want to check the LDAP status code
-(likely for `0`) on the `end` event to assert success.  LDAP search results
-can give you a lot of status codes, such as time or size exceeded, busy,
-inappropriate matching, etc., which is why this method doesn't try to wrap up
-the code matching.
+Responses inside callback of the `search` method are an `EventEmitter` where you will get a notification for
+each `searchEntry` that comes back from the server. You will additionally be able to listen for a `searchRequest`
+, `searchReference`, `error` and `end` event.
+`searchRequest` is emitted immediately after every `SearchRequest` is sent with a `SearchRequest` param. You can do op
+like `client.abandon` with `searchRequest.messageID` to abandon this search request. Note that the `error` event will
```
Comment (UziTech) [inline docs/client.md:283]:
```suggestion
`searchRequest` is emitted immediately after every `SearchRequest` is sent with a `SearchRequest` parameter. You can do operations
like `client.abandon` with `searchRequest.messageID` to abandon this search request. Note that the `error` event will
```

**Review by UziTech — CHANGES_REQUESTED** [review]
You can run `npm run lint` to fix the linting issues.

**Comment by zyf0330** [comment]
> You can run `npm run lint` to fix the linting issues.

Changes have been applied.

**Review by UziTech — APPROVED** [review]
LGTM

**Review by jsumners — APPROVED** [review]
Looks good to me.

**Comment by jsumners** [comment]
⚠️ This issue has been locked due to age. If you have encountered a recent
problem that seems to be covered by this issue, please open a new issue.

Please include a [minimal reproducible example](https://en.wikipedia.org/wiki/Minimal_reproducible_example)
when opening a new issue.

**Pull request merged**